pkg/kn/commands/revision/revision_list_test.go

```diff
@@ -71,17 +86,64 @@ func TestRevisionListDefaultOutput(t *testing.T) {
 		t.Errorf("Bad action %v", action)
 	}
-	testContains(t, output[0], []string{"SERVICE", "NAME", "AGE", "CONDITIONS", "READY", "REASON"}, "column header")
-	testContains(t, output[1], []string{"foo", "foo-abcd"}, "value")
-	testContains(t, output[2], []string{"bar", "bar-wxyz"}, "value")
+	commands.TestContains(t, output[0], []string{"SERVICE", "NAME", "AGE", "CONDITIONS", "READY", "REASON"}, "column header")
+	commands.TestContains(t, output[1], []string{"foo", "foo-abcd"}, "value")
+	commands.TestContains(t, output[2], []string{"bar", "bar-wxyz"}, "value")
+}
+
+func TestRevisionGetOneOutput(t *testing.T) {
+	revision := createMockRevisionWithParams("foo-abcd", "foo")
+	RevisionList := &v1alpha1.RevisionList{Items: []v1alpha1.Revision{*revision}}
+	action, output, err := fakeRevisionList([]string{"revision", "list", "foo-abcd"}, RevisionList)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if action == nil {
+		t.Errorf("No action")
+	} else if !action.Matches("list", "revisions") {
+		t.Errorf("Bad action %v", action)
+	}
+	commands.TestContains(t, output[0], []string{"SERVICE", "NAME", "AGE", "CONDITIONS", "READY", "REASON"}, "column header")
+	commands.TestContains(t, output[1], []string{"foo", "foo-abcd"}, "value")
 }
 
-func testContains(t *testing.T, output string, sub []string, element string) {
-	for _, each := range sub {
-		if !strings.Contains(output, each) {
-			t.Errorf("Missing %s: %s", element, each)
-		}
+func TestRevisionGetOutputWithTwoRevName(t *testing.T) {
+	t.Log("create a first revision name foo")
+	revision := createMockRevisionWithParams("foo-abcd", "foo")
+	RevisionList := &v1alpha1.RevisionList{Items: []v1alpha1.Revision{*revision}}
+	action, output, err := fakeRevisionList([]string{"revision", "list", "foo-abcd"}, RevisionList)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if action == nil {
+		t.Errorf("No action")
+	} else if !action.Matches("list", "revisions") {
+		t.Errorf("Bad action %v", action)
+	}
+	commands.TestContains(t, output[0], []string{"SERVICE", "NAME", "AGE", "CONDITIONS", "READY", "REASON"}, "column header")
+	commands.TestContains(t, output[1], []string{"foo", "foo-abcd"}, "value")
+
+	t.Log("create a second revision name bar")
+	revision = createMockRevisionWithParams("bar-abcd", "bar")
+	RevisionList = &v1alpha1.RevisionList{Items: []v1alpha1.Revision{*revision}}
+	action, output, err = fakeRevisionList([]string{"revision", "list", "bar-abcd"}, RevisionList)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if action == nil {
+		t.Errorf("No action")
+	} else if !action.Matches("list", "revisions") {
+		t.Errorf("Bad action %v", action)
 	}
+	commands.TestContains(t, output[0], []string{"SERVICE", "NAME", "AGE", "CONDITIONS", "READY", "REASON"}, "column header")
+	commands.TestContains(t, output[1], []string{"bar", "bar-abcd"}, "value")
+
+	t.Log("verify by passing two revision name to get command")
+	_, output, err = fakeRevisionList([]string{"revision", "list", "foo-abcd", "bar-abcd"}, RevisionList)
```

Comment:
above test lines in this test are redundant, thats already being tested in `TestRevisionGetOneOutput`, if we are checking for error message here, no need to populate the `RevisionList`, it can be empty.

Reply:
Updated